Basic.wl

```diff
@@ -187,10 +187,69 @@ ImprisonmentExpressions[template_List]:= Cases[template, x_ \[Element] set_ ,Inf
 ValueRestrictions[imprisonmentExpression_]:=
  Apply[Or,imprisonmentExpression[[1]] == #&/@ imprisonmentExpression[[2]]];
 
+RemoveContainedSubsSets[allSubSet_List, subSet_List] :=
+ Module[{listToRemove, result},
+  listToRemove = ContainsAll[#, subSet] && subSet =!= # & /@ allSubSet;
+  result = MapThread[If[! #2, #1, ## &[]] &, {allSubSet, listToRemove}]
+]
+
+MinimizeSets[subSets_List] :=
```

Comment:
This function is really hard to read and reason about.
A few refactoring tips for it: 
* Rename short-named symbols like `p0`, `p1` and `v` to have meaningful names. It is impossible to someone reading this code for the first time to understand what those symbols stand for.
* Extract long expressions to functions with meaningful names. Specially if they are already anonymous functions. For instance, I'm quite positive this anonymous function `And @@ # & /@ (Or @@ Map[If[#[[2]] === 0, \[Not] #[[1]], #[[1]]] &` could be extracted to a named function with a name that makes it easier for the reader to understand what is going on.
* Still about `And @@ # & /@ (Or @@ Map[If[#[[2]] === 0, \[Not] #[[1]], #[[1]]]` : there is no problem using Mathematica's syntactic sugar like `/@` or `@@` or `#`, but keep in mind that mixing a lot of those often makes the code confusing and less readable.
* Watch out for misleading indentation. It took me quite a while to begin grasping what the whole `p0 = ...` statement meant, partially because of what I mentioned :point_up: , partly because the indentation is misleading.
* Avoid temporal coupling caused by bad `module` usages. If you extracted the whole `p0` definition to a function, you could define whataver `p1` means in terms of this new function, instead of depending on `p0` being calculated first. You can compare how the [new ColorBlindTemplate](https://github.com/mverardo/CATemplates/blob/f6e52074ff78b773c43cf46d2d8262f347afb22d/TemplateGeneration/ColorBlindTemplate.wl) is far easier to grasp than the [old one](https://github.com/mverardo/CATemplates/blob/df645c94594115725cddbbac8f96bc7b80dd67a6/TemplateGeneration/ColorBlindTemplate.wl) because I've replace some of the temporal coupling with clearer function definitions.
* Avoid nested `If` and `Which` statements. If you really can't find a way to avoid 'Which', use functions with pattern matching instead.